### PR TITLE
feature: Generate error codes from source

### LINKED ISSUE #162: ErrorCode has fallen out of sync with upstream
We should add test that asserts all errors defined upstream are present in `ErrorCode` — or, even better — generate the kio definition based on the upstream definition.

### PR DESCRIPTION
**Pull request by aiven-anton**
Fixes #162.

### CODE REVIEW

File: java_tester/src/main/java/io/aiven/kio/java_tester/JavaTester.java [hunk 1]
```diff
@@ -23,7 +25,28 @@ public class JavaTester {
...
     public static void main(String[] args) throws Exception {
+        if (Arrays.stream(args).anyMatch(CMD_PRINT_ERROR_CODES::equals)) {
+            System.err.println("Printing error codes");
+
+            for (Errors error : Errors.values()) {
+                System.out.print(error.code());
+                System.out.print(" ");
+                System.out.print(error.name());
+                System.out.print(" ");
+                // https://github.com/a0x8o/kafka/blob/54eff6af115ee647f60129f2ce6a044cb17215d0/clients/src/main/java/org/apache/kafka/common/protocol/Errors.java#L521
```
Comment (antonagestam) [inline java_tester/src/main/java/io/aiven/kio/java_tester/JavaTester.java:39]: Link to wrong repository

**Comment by aiven-anton** [comment]
The packaging metadata check is bumping into this issue: https://github.com/actions/setup-python/issues/886

The other checks were luckier and reused cached Python 3.12.3.

**Comment by aiven-anton** [comment]
Packaging metadata is failing due to upstream breakage in twine: https://github.com/pypa/twine/issues/977

**Review by giuseppelillo — APPROVED** [review]
Nice one

**Pull request merged**